ansible_builder/main.py

```diff
@@ -191,21 +217,12 @@ def create_folder_copy_files(self):
 
         os.makedirs(self.build_context, exist_ok=True)
 
-        for item in CONTEXT_FILES:
+        for item, new_name in CONTEXT_FILES.items():
             requirement_path = self.definition.get_dep_abs_path(item)
             if requirement_path is None:
                 continue
-            if item == 'python':
-                continue  # will be put into combined requirements.txt
-            dest = os.path.join(self.build_context, os.path.basename(requirement_path))
-            exists = os.path.exists(dest)
-            do_copy = True
-            if exists:
-                do_copy = not filecmp.cmp(requirement_path, dest, shallow=False)
-                if do_copy:
-                    print(MessageColors.WARNING + 'File {} had modifications and will be rewritten'.format(dest) + MessageColors.ENDC)
-            if do_copy:
-                shutil.copy(requirement_path, self.build_context)
+            dest = os.path.join(self.build_context, new_name)
+            copy_file(requirement_path, dest)
```

Comment:
Coding practices for copy/write had diverged in a few places, so I tried to fully finish shared methods for `copy_file` and `write_file`